x-pack/metricbeat/module/azure/billing/service.go

```diff
@@ -40,26 +45,52 @@ func NewService(config azure.Config) (*UsageService, error) {
 	if err != nil {
 		return nil, err
 	}
-	forcastsClient := consumption.NewForecastsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
-	usageDetailsClient := prevConsumption.NewUsageDetailsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
 
-	forcastsClient.Authorizer = authorizer
+	forecastsClient := consumption.NewForecastsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
+	usageDetailsClient := consumption.NewUsageDetailsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
+
+	forecastsClient.Authorizer = authorizer
 	usageDetailsClient.Authorizer = authorizer
-	service := &UsageService{
+
+	service := UsageService{
 		usageDetailsClient: &usageDetailsClient,
-		forcastsClient:     &forcastsClient,
+		forecastsClient:    &forecastsClient,
 		context:            context.Background(),
 		log:                logp.NewLogger("azure billing service"),
 	}
-	return service, nil
+
+	return &service, nil
 }
 
-// GetForcast
-func (service *UsageService) GetForcast(filter string) (consumption.ForecastsListResult, error) {
-	return service.forcastsClient.List(service.context, filter)
+// GetForecast fetches the forecast for the given filter.
+func (service *UsageService) GetForecast(filter string) ([]consumption.Forecast, error) {
+	response, err := service.forecastsClient.List(service.context, filter)
+	if err != nil {
+		switch response.StatusCode {
+		case 404:
+			// Forecast API returns 404 when the subscription does not support forecasts.
+			// For example, at the time of writing, forecasts are only available to
+			// enterprises subscriptions:
+			//
+			// "[Forecasts API] Provides operations to get usage forecasts for Enterprise
+			// Subscriptions." [1]
+			//
+			// [1]: https://docs.microsoft.com/en-us/rest/api/consumption/
+			service.log.Warnf(
```

Comment:
do you think it's worth providing a URL in the warning log event for customers to be able to follow up on exactly why their subscription is not supported? or just something to point the customer to the right place, as opposed to them potentially reaching out to our support team...

Reply:
I updated the message to include the URL.